napari/components/_layer_slicer.py

```diff
@@ -0,0 +1,150 @@
+from __future__ import annotations
+
+import logging
+from concurrent.futures import Executor, Future, ThreadPoolExecutor
+from threading import RLock
+from typing import Dict, Iterable, Optional, Tuple
+
+from napari.components import Dims
+from napari.layers import Layer
+from napari.utils.events.event import EmitterGroup, Event
+
+logger = logging.getLogger("napari.components._layer_slicer")
+
+
+class _LayerSlicer:
+    """
+    High level class to control the creation of a slice (via a slice request),
+    submit it (synchronously or asynchronously) to a thread pool, and emit the
+    results when complete.
+
+    Attributes
+    ----------
+    events : napari.utils.events.event.EmitterGroup
+        vispy event emitter
+    _executor : concurrent.futures.ThreadPoolExecutor
+        manager for the slicing threading
+    _layers_to_task : dict
```

Comment:
Optional: I'd prefer to not document private attributes in the class docstring (though we do that elsewhere). Instead, I think documenting them in `__init__` might be better.